deforcefields/tests/test_deforcefields.py

```diff
@@ -89,3 +89,91 @@ def test_fails_unsupported_chemistry():
         ).create_interchange(
             Molecule.from_mapped_smiles("[H:3][C:1]#[C:2][H:4]").to_topology()
         ).to_openmm(combine_nonbonded_forces=False)
+
+
+@pytest.mark.parametrize(
+    "forcefield, ref_energy",
+    [
+        pytest.param("de-force-1.0.2.offxml", 13.601144438830156, id="No constraints"),
+        pytest.param(
+            "de-force_unconstrained-1.0.2.offxml", 13.605201859835375, id="Constraints"
+        ),
+    ],
+)
+def test_energy_no_sites(forcefield, ref_energy, ethanol_with_charges):
+    """
+    Test calculating the single point energy of ethanol using constrained and unconstrained DE-FF with pre-computed
+    partial charges from openeye.
+    """
+
+    ff = ForceField(forcefield, load_plugins=True)
+
+    interchange = ff.create_interchange(
+        topology=ethanol_with_charges.to_topology(),
+        charge_from_molecules=[ethanol_with_charges],
+    )
+
+    found_energy = get_openmm_energies(
+        interchange,
+        combine_nonbonded_forces=False,
+    ).total_energy.m_as(kj_mol)
+
+    assert found_energy == pytest.approx(ref_energy)
+
+
+def evaluate_water_energy_at_distances(
+    force_field: ForceField,
+    distances: list[float],
+) -> list[Quantity]:
+    """
+    Evaluate a water dimer at specified distances (in Angstrom).
+
+    Taken from smirnoff_plugins.utilities.openmm, which collates virtual particles
+    between molecules. Interchange (with OpenMM) puts all virtual sites at the END
+    of the topology; mismatching these causes NaNs.
+    """
+
+    water = Molecule.from_smiles("O")
+    water.generate_conformers(n_conformers=1)
+    topology = Topology.from_molecules([water, water])
+    topology.box_vectors = unit.Quantity(numpy.eye(3) * 20, unit.nanometer)
+
+    energies = list()
+
+    for distance in distances:
+        topology.set_positions(
+            numpy.vstack(
+                [
+                    water.conformers[0],
+                    water.conformers[0] + Quantity(
+                        numpy.array([distance, 0, 0]),
+                        "angstrom",
+                    ),
+                ],
+            ),
+        )
+
+        energies.append(
+            get_openmm_energies(
+                force_field.create_interchange(topology),
+                combine_nonbonded_forces=False,
+            ).total_energy
+        )
+
+    return energies
+
+def test_energy_sites():
+    """
+    Test calculating the energy for a system with two waters with virtual sites at set distances.
+    """
+
+    ff = ForceField("de-force-1.0.2.offxml", load_plugins=True)
+
+    found_energies = evaluate_water_energy_at_distances(
+        force_field=ff,
+        distances=[2, 3, 4],
+    )
+    ref_energies = [1005.0846252441406, 44.696786403656006, 10.453390896320343]
+
+    for found, ref in zip(found_energies, ref_energies):
+        assert found.m_as(kj_mol) == pytest.approx(ref, rel=1e-3)
```

Comment:
Note this is a fairly loose tolerance; I was getting errors like these if I used the defaults. How exactly these energies have drifted slightly from their artisan hand-crafted values a couple of years back, I'm not sure 


```
$ python -m pytest -v deforcefields/tests/test_deforcefields.py::test_energy_sites | grep "E    "
E       assert 1005.0861476907567 == 1005.0846252441406 ± 1.0e-03
E
E         comparison failed
E         Obtained: 1005.0861476907567
E         Expected: 1005.0846252441406 ± 1.0e-03
E       assert 44.69819545912324 == 44.696786403656006 ± 4.5e-05
E
E         comparison failed
E         Obtained: 44.69819545912324
E         Expected: 44.696786403656006 ± 4.5e-05
E       assert 10.454626091527247 == 10.453390896320343 ± 1.0e-05
E
E         comparison failed
E         Obtained: 10.454626091527247
E         Expected: 10.453390896320343 ± 1.0e-05

Reply:
I think the original values were calculated using the toolkit pre-interchange but maybe its time to do them by hand to double check, for now though this is a great match!